main/src/library/DelayMap.flix

```diff
@@ -20,6 +20,11 @@ pub enum DelayMap[k, v] {
     case DMap(RedBlackTree[k, Lazy[v]])
 }
 
+instance Iterable[DelayMap[k]] {
+    pub def iterator(t: DelayMap[k, a]): Iterator[a] & Impure =
+        t |> DelayMap.valuesOf |> List.toIterator
```

Comment:
This solution is not so good because it actually means you convert the entire map to a list. This defeats the purpose of an iterator which is supposed to be "cheap". Thus some other mechanism is needed.

Reply:
@jaschdoc Do you have any ideas of how to approach this?

Reply:
To expand on the problem, imagine the tree has 1_000_000 elements. Then you construct a list of 1_000_000 elements. Then you use the iterator to only get the first element. In this case a lot of work has been wasted and this is what we try to avoid with iterators. Makes sense? :)

Reply:
ah, I see. I have done this other places too. Should I change these to do it more directly?

Reply:
> To expand on the problem, imagine the tree has 1_000_000 elements. Then you construct a list of 1_000_000 elements. Then you use the iterator to only get the first element. In this case a lot of work has been wasted and this is what we try to avoid with iterators. Makes sense? :)

Yes, makes a lot of sense. I just didn't think of it before :)

Reply:
> ah, I see. I have done this other places too. Should I change these to do it more directly?

Yes

Reply:
Also note here that the values of `DelayMap` are not yet evaluated i.e. the type of any value `v` is `Lazy[v]`. Maybe the `next` function of the iterator in this case should force the current element?